common/canonical_serialization/src/simple_deserializer.rs

```diff
@@ -59,26 +59,6 @@ impl<'a> CanonicalDeserializer for SimpleDeserializer<'a> {
         Ok(b != 0)
     }
 
-    fn decode_btreemap<K: CanonicalDeserialize + std::cmp::Ord, V: CanonicalDeserialize>(
```

Comment:
Why is this file changed? It seems like code got reorganized. Could you revert it?

Reply:
@weilei0107 it's on purpose, just took the opportunity to improve readability and consistency (implementation order to match function definition order on the trait).

Reply:
Perhaps we should have fixed the definition within the trait. Now this is no longer alphabetically ordered :(

Reply:
@davidiw good point, but I guess there're two different opinions here: "alphabetical" vs "conceptual". 
For instance, the creator of a `trait` might decided that numbers should be grouped even if alphabetically there is some distance. Thus, 
`decode` i8, i32, i64, i128, u8, u32, u64, u128, string 
might be better conceptually than the alphabetical ordering 
`decode` i128, i8, i32, i64, string, u128, u8, u32, u64